### PR TITLE
[18.0][MIG] account_invoice_start_end_dates: Migration to 18.0

### PR DESCRIPTION
**Pull request by xaviedoanhduy**
supersedes: https://github.com/OCA/account-closing/pull/307

### CODE REVIEW

File: account_invoice_start_end_dates/tests/test_invoice_start_end_dates.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                            "account_id": self.account_revenue.id,
+                            "start_date": self._date("01-01"),
+                            "end_date": self._date("06-30"),
+                        },
+                    ),
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": self.env.ref("product.product_product_5").id,
```
Comment (ajaniszewska-dev) [inline account_invoice_start_end_dates/tests/test_invoice_start_end_dates.py:84]: looking at this test case i guess we test positive case here - all previous lines have dates set and we don't expect any error to be raised for product_product_5 because probably this product has no flag set - i'd maybe add some comments in this case + test negative case (where error should be raised)?

**Review by ajaniszewska-dev — APPROVED** [review]
LG, i'd slightly improve test

File: account_invoice_start_end_dates/tests/test_invoice_start_end_dates.py (hunk 1 above)
Comment (xaviedoanhduy) [reply account_invoice_start_end_dates/tests/test_invoice_start_end_dates.py:84]:
```
if moveline.start_date and not moveline.end_date:
    ...
if moveline.end_date and not moveline.start_date:
    ...
if (
    moveline.end_date
    and moveline.start_date
    and moveline.start_date > moveline.end_date
):
```
hi @ajaniszewska-dev, I had the same question as you, I'm not sure what the author of this module intended, because the above conditions require at least 1 value to be assigned to `start_date` and `end_date`, so if no value is inputed, no error will be raised. I tried adding a filter in the lookup function like:
```
def _check_start_end_dates(self):
    for moveline in self.filtered("must_have_date"):
         ...
```
then the error will be raised, and I wonder if that change affects the original purpose of this module?

Comment (ajaniszewska-dev) [reply account_invoice_start_end_dates/tests/test_invoice_start_end_dates.py:84]: @alexis-via  probably your input here would be valuable.

**Comment by alexis-via** [comment]
I tried your PR, but faced several issues:

mig_closing_popup

and also an issue in product form view (at least when using with intrastat_base... it's kind of a decision if we should create groups inside the native group name="accounting" (choice made by intrastat_base) or put fields directly in it.

closing_product_form_view

So, for these 2 reasons, I decided to make another PR https://github.com/OCA/account-closing/pull/313

File: account_invoice_start_end_dates/tests/test_invoice_start_end_dates.py (hunk 1 above)
Comment (alexis-via) [reply account_invoice_start_end_dates/tests/test_invoice_start_end_dates.py:84]: I added the negative tests in my migration PR https://github.com/OCA/account-closing/pull/313
